storage/replica.go

```diff
@@ -513,8 +513,12 @@ func (r *Replica) requestLeaderLease(timestamp roachpb.Timestamp) <-chan *roachp
 		// Send result of leader lease to all waiter channels.
 		r.mu.Lock()
 		defer r.mu.Unlock()
-		for _, llChan := range r.mu.llChans {
-			llChan <- pErr
+		for i, llChan := range r.mu.llChans {
+			if i == 0 {
+				llChan <- pErr
+			} else {
+				llChan <- protoutil.Clone(pErr).(*roachpb.Error)
```

Comment:
Geez, I think you're right. Will write a test.

On Mon, Apr 18, 2016, 15:27 Ben Darnell <notifications@github.com> wrote:

> In storage/replica.go
> <https://github.com/cockroachdb/cockroach/pull/6111#discussion_r60056865>:
>
> > @@ -513,8 +513,12 @@ func (r *Replica) requestLeaderLease(timestamp roachpb.Timestamp) <-chan *roachp
> >  		// Send result of leader lease to all waiter channels.
> >  		r.mu.Lock()
> >  		defer r.mu.Unlock()
> > -		for _, llChan := range r.mu.llChans {
> > -			llChan <- pErr
> > +		for i, llChan := range r.mu.llChans {
> > +			if i == 0 {
> > +				llChan <- pErr
> > +			} else {
> > +				llChan <- protoutil.Clone(pErr).(*roachpb.Error)
>
> Won't this panic if pErr is nil? Looks like we need some test coverage of
> multiple operations waiting on the same lease request.
>
> —
> You are receiving this because you modified the open/close state.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/6111/files/e7d357b97777afa2c78a0321ef57eb1deeb3fe06#r60056865>
>
-- 

-- Tobias